src/orange/univ/detail/LazySenseCalculator.hh

```diff
@@ -0,0 +1,163 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021-2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/univ/detail/LazySenseCalculator.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Assert.hh"
+#include "corecel/cont/Range.hh"
+#include "corecel/cont/Span.hh"
+#include "orange/OrangeTypes.hh"
+#include "orange/surf/LocalSurfaceVisitor.hh"
+#include "orange/univ/detail/Types.hh"
+
+#include "SurfaceFunctors.hh"
+#include "../VolumeView.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate senses with a fixed particle position.
+ *
+ * This is an implementation detail used in initialization *and* complex
+ * intersection.
```

Comment:
Add more description here. Note also you're going from `senses` to a single sense on one face.